scipy/cluster/tests/test_hierarchy.py

```diff
@@ -211,12 +211,12 @@ def test_linkage_cophenet_tdist_Z_Y(self, xp):
 
 class TestMLabLinkageConversion:
 
-    @skip_if_array_api
-    def test_mlab_linkage_conversion_empty(self):
+    @array_api_compatible
+    def test_mlab_linkage_conversion_empty(self, xp):
         # Tests from/to_mlab_linkage on empty linkage array.
-        X = np.asarray([])
-        assert_equal(from_mlab_linkage([]), X)
-        assert_equal(to_mlab_linkage([]), X)
+        X = xp.asarray([], dtype=xp.float64)
+        xp_assert_equal(from_mlab_linkage(X), X)
+        xp_assert_equal(to_mlab_linkage(X), X)
```

Comment:
Maybe we want to also still test the input `[]` but I don't see that regressing without a bunch of other failures too.

Reply:
Hmm, I recall some discussions where there was a concern about trying to preserve array-like (input) -> NumPy array (output), but I'm assuming you considered that. I guess the test may get a little messy if you do both array-like and array API support, but might be worth doing.

Reply:
> I recall some discussions where there was a concern about trying to preserve array-like (input) -> NumPy array (output), but I'm assuming you considered that

Yeah, me too; if we don't want to lose _any_ test coverage, then we can reinstate the original test and mark it as `skip_if_xp_not_numpy`. I'm not convinced that this would ever break without us noticing elsewhere but whatever you think is best practice :+1:

Reply:
I think we should decide what to do here and document it. I would say:

- When a parameter takes (e.g.) either an `ndarray` or an `int`, we should have tests for both types, for all backends.
- When a parameter takes just an `ndarray` (but e.g. `int`s happen to work with NumPy), we do not support array-likes for alternative backends. `xp.asarray` should be used in all tests, and a singular separate test, just for NumPy, should check array-like input.

Does that sound right @rgommers ?

Reply:
The parameter is documented to accept an `ndarray`, so I'm happy to drop the array-like test here. Resolving.